uwrt_utils/include/uwrt_utils/uwrt_can.h

```diff
@@ -0,0 +1,135 @@
+#pragma once
+
+#include <linux/can.h>
+#include <linux/can/raw.h>
+#include <net/if.h>
+
+#include <chrono>
+#include <cstring>
+#include <map>
+#include <mutex>
+#include <stdexcept>
+#include <thread>
+#include <vector>
+
+namespace uwrt_utils {
+
+class UWRTCANWrapper {
+ public:
+  explicit UWRTCANWrapper() = default;
+  explicit UWRTCANWrapper(std::string name, std::string interface_name, bool rcv_big_endian,
+                          int thread_sleep_millis = 1);
+  explicit UWRTCANWrapper(const UWRTCANWrapper& to_copy) = delete;
+  // NOLINTNEXTLINE(performance-noexcept-move-constructor, bugprone-exception-escape)
+  explicit UWRTCANWrapper(UWRTCANWrapper&& to_move);
+
+  ~UWRTCANWrapper();
+
+  UWRTCANWrapper& operator=(const UWRTCANWrapper& to_copy) = delete;
+  // NOLINTNEXTLINE(performance-noexcept-move-constructor, bugprone-exception-escape)
+  UWRTCANWrapper& operator=(UWRTCANWrapper&& to_move);
+
+  void init(const std::vector<uint32_t>& ids);
+
+ private:
+  // description name for this wrapper
+  std::string name_;
+
+  // variables needed for can socket
+  std::string interface_name_;
+  int socket_handle_{};
+  struct sockaddr_can sock_addr_ {};
+  struct ifreq ifr_ {};
+
+  // variables for wrapper
+  bool initialized_{false};
+  int rcv_endianness_{__BYTE_ORDER__};
+  std::thread read_thread_;
+  volatile bool read_thread_running_{};
+  std::map<canid_t, struct can_frame> recv_map_;
+  std::timed_mutex recv_map_mtx_;
+  std::chrono::milliseconds thread_sleep_millis_{};
+  static constexpr std::chrono::milliseconds MUTEX_LOCK_TIMEOUT{1};
+
+  // read function to be run in the thread
+  void readSocketTask();
+
+  // function to get swap endianness
+  template <class T>
+  T correctEndianness(const T& data) {
+    // if endianness is the same, just return the data back
+    if (rcv_endianness_ == __BYTE_ORDER__) {
+      return data;
+    }
+
+    // if not, we need to swap endianess
+    T swp_data;
```

Comment:
try to avoid abbreviating words unless they are reaaaalllllyyyy long
`swapped_data`, `swapped_data_ptr` , `original_data_ptr`